.github/workflows/maven-ci.yml

```diff
@@ -0,0 +1,108 @@
+name: OTP CI Build
+
+# TODO: Perhaps reinstate, but doesn't allow for build/test of any branch.
+# On [push, pull_request] causes double-builds when creating PRs.
+# But triggering on push only will miss pull requests from outside authors.
+# The push event's ref is the name of the pushed branch;
+# The pull_request event's branch name is the merge target branch.
+#on:
+#  push:
+#    branches:
+#      - master
+#      - dev-1.x
+#      - dev-2.x
+#      - ibi-dev
+#  pull_request:
+#    branches:
+#      - master
+#      - dev-1.x
+#      - dev-2.x
+#      - ibi-dev
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      JAVA_OPTS: -Xmx2g
+      # TODO Remove if not required.
+      OSSRH_JIRA_USERNAME: ${{ secrets.OSSRH_JIRA_USERNAME }}
+      OSSRH_JIRA_PASSWORD: ${{ secrets.OSSRH_JIRA_PASSWORD }}
+      GPG_KEY_NAME: ${{ secrets.GPG_KEY_NAME }}
+      GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+
+    steps:
+      # Starting in v2.2 checkout action fetches all tags when fetch-depth=0, for auto-versioning.
+      - uses: actions/checkout@v2.3.2
+        with:
+          fetch-depth: 0
+      # Java setup step completes very fast, no need to run in a preconfigured docker container.
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      # Retain the local Maven repository to speed up builds.
+      - name: Setup Maven Cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: ~/.m2
+          # TODO Fix error message 'Cache not found for input keys: maven-local-repo'
+          key: maven-local-repo
+      # So that the branch name can be reference via $GITHUB_HEAD_REF_SLUG.
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v3.x
+      # Define AWS profile credentials for use by subsequent steps.
+      - name: Add profile credentials to ~/.aws/credentials
+        if: github.event_name == 'push'
+        run: ./scripts/add-aws-credentials.sh
```

Comment:
I don't think this step is needed since we're using the `jakejarvis/s3-sync-action` action.

Reply:
Step removed.

Reply:
Step removed.

Reply:
Unresolving. Giving AWS credentials to a 3rd party is an unnecessary security risk.

Reply:
Now using script and AWS creds are only available to this step.